apps/condo/domains/ticket/tasks/sendCreateTicketNotificationsToEmployees.js

```diff
@@ -0,0 +1,140 @@
+            ticketVisibilityType: PROPERTY_TICKET_VISIBILITY,
+        },
+    })
+    const employeesWithPropertyAndSpecializationTicketVisibility = await find('OrganizationEmployee', {
+        ...baseEmployeesQuery,
+        role: {
+            ticketVisibilityType: PROPERTY_AND_SPECIALIZATION_VISIBILITY,
+        },
+    })
+
+    const ticketProperty = get(newItem, 'property')
+    const ticketCategory = get(newItem, 'categoryClassifier')
+
+    const organizationPropertyScopes = await find('PropertyScope', {
+        organization: { id: organizationId, deletedAt: null },
+        deletedAt: null,
+    })
+    const employeeSpecializations = await find('OrganizationEmployeeSpecialization', {
+        employee: { id_in: employeesWithPropertyAndSpecializationTicketVisibility.map(obj => obj.id), deletedAt: null },
+        specialization: { id: ticketCategory, deletedAt: null },
+        deletedAt: null,
+    })
+    const employeesWithMatchedCategory = await find('OrganizationEmployee', {
+        id_in: compact(employeeSpecializations.map(obj => get(obj, 'employee', null))),
+        deletedAt: null,
+    })
+
+    // all employees with organization ticket visibility, because they are visible all organization tickets
+    employeeUsersToSendNotification.push(...(employeesWithOrganizationTicketVisibility.map(employee => employee.user)))
+    // assignee and executor of ticket (ASSIGNED_TICKET_VISIBILITY)
+    employeeUsersToSendNotification.push(...[newItem.assignee, newItem.executor])
+
+    const isDefaultScopeExists = organizationPropertyScopes.find(scope => scope.hasAllProperties && scope.hasAllEmployees)
+
+    if (isDefaultScopeExists) {
+        employeeUsersToSendNotification.push(...(employeesWithPropertyTicketVisibility.map(employee => employee.user)))
+        employeeUsersToSendNotification.push(...(employeesWithMatchedCategory.map(employee => employee.user)))
+    } else {
+        const propertyScopeProperties = await find('PropertyScopeProperty', {
+            propertyScope: { id_in: organizationPropertyScopes.map(scope => scope.id), deletedAt: null },
+            property: { id: ticketProperty, deletedAt: null },
+            deletedAt: null,
+        })
+        const scopesWithTicketProperty = organizationPropertyScopes.filter(
+            scope => propertyScopeProperties.find(obj => obj.propertyScope === scope.id) || scope.hasAllProperties
+        )
+
+        if (scopesWithTicketProperty.find(scope => scope.hasAllEmployees)) {
+            employeeUsersToSendNotification.push(...(employeesWithPropertyTicketVisibility.map(employee => employee.user)))
+            employeeUsersToSendNotification.push(...(employeesWithMatchedCategory.map(employee => employee.user)))
+        } else {
+            const propertyScopeOrganizationEmployees = await find('PropertyScopeOrganizationEmployee', {
+                deletedAt: null,
+                propertyScope: { id_in: scopesWithTicketProperty.map(scope => scope.id) },
```

Comment:
perhaps `propertyScope` also needs `deletedAt` filter

Reply:
and test for that cases